frontend/src/components/custom-table.js

```diff
@@ -0,0 +1,177 @@
+import React from "react";
+import Table from "react-table";
+import selectTableHOC from "react-table/lib/hoc/selectTable";
+
+const SelectTable = selectTableHOC(Table);
+
+export class CustomTable extends React.Component {
```

Comment:
Make a functional component. Add doc strings to this component as well as all subfunctions you've defined.